### PR TITLE
feature/fine-tune text input

### PR DESCRIPTION
**Pull request by nevyangelova**
#### Summary
fine tuning text input field according to design notes. @michelengelen before i move on with the design review I would like to discuss a style change. I would propose moving the Label component back to the input root. IMO at the moment they share too many props and logic between each other to make sense to split. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37259

### CODE REVIEW

File: src/components/text-input/Input.root.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ const InputRoot = styled.input<PInputRoot>(
 
         &::placeholder {
             color: transparent;
-            transition: color 200ms ease-in;
+            transition: color 100ms ease-in;
```
Comment (michelengelen) [inline src/components/text-input/Input.root.tsx:17]: we now have theme-variables for that

**Review by michelengelen — APPROVED** [review]
LGTM ... just a smallish remark

File: src/components/text-input/TextInput.root.tsx [hunk 2]
```diff
@@ -74,6 +92,34 @@ const TextInputRoot = styled.div<PTextInputRoot>(
...
+                )};
+                ${applyTextStyles({
+                    size: TEXT_INPUT_VALUES_MAPPING[size].labelSize,
+                    inheritLineHeight: true,
+                })};
+
+                transition: all ${animation.fastest}ms linear;
+            }
+
+            .clear {
```
Comment (michelengelen) [inline src/components/text-input/TextInput.root.tsx:119]: the trailing icon will have different options (basically just running a callback function, or be decorative). Either way the naming is a bit awkward, when you somehow imply a function to it, which it can, but not has to be doing. 

e.g. it could also be a trigger to perform a search.

Better naming would be `.trailing-icon` and instead of `isClearable` just name it `isTrailingIconFunctional`? Open to suggestions!

File: src/components/text-input/TextInput.root.tsx [hunk 3]
```diff
@@ -57,12 +75,12 @@ const TextInputRoot = styled.div<PTextInputRoot>(
         return css`
             ${actionStyles};
             color: ${colors.text};
+            box-sizing: content-box;
```
Comment (michelengelen) [inline src/components/text-input/TextInput.root.tsx:77]: why is this needed?

File: src/components/text-input/TextInput.root.tsx [hunk 4]
```diff
@@ -49,6 +52,20 @@ const TextInputRoot = styled.div<PTextInputRoot>(
...
+                      border-color: ${colors.active};
+                      background-color: ${setAlpha(colors.active, 0.16)};
+                  }
+
+                  input:focus + span {
+                      background-color: ${backgroundColor};
+                      ${animatedLabel
+                          ? `transform: translate(${LABEL_POSITIONS[size]}) scale(0.7); color: ${colors.active};`
+                          : 'transform: scale(0); opacity: 0;'};
+                  }
```
Comment (michelengelen) [inline src/components/text-input/TextInput.root.tsx:68]: you can use `css\`\`` here as well

File: src/components/text-input/TextInput.root.tsx [hunk 5]
```diff
@@ -49,6 +53,20 @@ const TextInputRoot = styled.div<PTextInputRoot>(
...
+                  &:active {
+                      border-color: ${colors.active};
+                      background-color: ${setAlpha(colors.active, 0.16)};
+                  }
+
+                  input:focus + span {
+                      background-color: ${backgroundColor};
+                      ${animatedLabel
+                          ? `transform: translate(${LABEL_POSITIONS[size]}) scale(0.7); color: ${colors.active};`
+                          : 'transform: scale(0); opacity: 0;'};
```
Comment (michelengelen) [inline src/components/text-input/TextInput.root.tsx:67]: you can use the css function here as well

File: src/components/text-input/TextInput.tsx [hunk 6]
```diff
@@ -42,6 +42,8 @@ const TextInput: React.FC<PTextInput> = ({
         animatedLabel,
         backgroundColor,
         width,
+        onClear,
```
Comment (michelengelen) [inline src/components/text-input/TextInput.tsx:46]: this should probably be named differently as well, since the trailingIcon will not always be a clear-icon, but instead can have a different action added to it.

The name is a bit clunky, but `handleTrailingIconClick` or `onTrailingIconClick` should be more descriptive of what it is supposed to do.

**Review by michelengelen — CHANGES_REQUESTED** [review]

File: src/components/text-input/TextInput.root.tsx (hunk 3 above)
Comment (nevyangelova) [reply src/components/text-input/TextInput.root.tsx:77]: when we discussed the paddings with Michael there was a mismatch with the designs as in the size they wanted didn't include the borders. Also when we use tab-focus the position of the element switch slightly because we are adding the extra border. In order to keep the consistent padding with the values we specified and to prevent the slight shift, this fixes both issues.

File: src/components/text-input/TextInput.tsx (hunk 6 above)
Comment (nevyangelova) [reply src/components/text-input/TextInput.tsx:46]: This was another discussion I had with @michaelgamble that the trailing icon will always be a clearing one and would also need to appear only if there is a value applied to it. Maybe he can advice here too?

Comment (michelengelen) [reply src/components/text-input/TextInput.tsx:46]: this is kinda against our generic approach, but fair eonugh ... IMO this is just a small change with a potentially big impact! :D

File: src/components/text-input/TextInput.stories.mdx [hunk 7]
```diff
@@ -188,18 +188,18 @@ export const ControlledTextInput = (args) => {
 
 export const noTrailingIconArgs = {
     ...defaultArgs,
-    trailingIcon: 'none',
+    trailingIcon: '',
```
Comment (michelengelen) [inline src/components/text-input/TextInput.stories.mdx:191]: nit: an empty string is not a valid option as `null` would be for this prop, but since it serves the purpose this is fine for now!

**Review by michelengelen — APPROVED** [review]
nitpicky, but LGTM otherwise! ✨

**Comment by michelengelen** [comment]
@michaelgamble kind reminder to review! 🏢
